contracts/zero-ex/contracts/src/features/SignatureValidator.sol

```diff
@@ -160,12 +167,18 @@ contract SignatureValidator is
             uint8 v = uint8(signature[0]);
             bytes32 r = signature.readBytes32(1);
             bytes32 s = signature.readBytes32(33);
-            recovered = ecrecover(
-                hash,
-                v,
-                r,
-                s
-            );
+            if (v < 27) {
+                // Handle clients that encode v as 0 or 1.
+                v += 27;
+            }
```

Comment:
It looks like the valid values for `v` are `{27,28}`, after adjustment. Not sure if there's a strong security reason for verifying, but we could also assert this range.

Reply:
Anything outside that range should just cause `ecrecover()` to fail.

Reply:
I honestly think we should remove the `s` and `r` range checks. Signatures don't get included in any of our hashes, so replay isn't an issue. Plus, allowing `v` to be `0-1` or `27-28` introduces a new malleability since you can just take the same signature, flip `v` to `0-1`, and it'll be made valid.

Reply:
Oh interesting, that makes sense. Yeah I'm down to nix them. Perhaps just run by Paul real quick, since it was his recommendation.

Reply:
Paul's recco is to actually just drop the parity `v` support.